lib/widgets/address_item.dart

```diff
@@ -0,0 +1,164 @@
+import 'package:my_wit_wallet/constants.dart';
+import 'package:my_wit_wallet/theme/colors.dart';
+import 'package:my_wit_wallet/theme/extended_theme.dart';
+import 'package:my_wit_wallet/util/get_localization.dart';
+import 'package:my_wit_wallet/util/storage/database/account.dart';
+import 'package:my_wit_wallet/widgets/PaddedButton.dart';
+import 'package:my_wit_wallet/util/extensions/string_extensions.dart';
+import 'package:my_wit_wallet/widgets/snack_bars.dart';
+import 'package:my_wit_wallet/util/extensions/num_extensions.dart';
+
+class AddreessItem extends StatefulWidget {
+  final bool isLastItem;
+  final Account account;
+  final String? currentAddress;
+  const AddreessItem(
+      {Key? key,
+      required this.isLastItem,
+      required this.account,
+      required this.currentAddress})
+      : super(key: key);
+
+  @override
+  State<AddreessItem> createState() => AddreessItemState();
+}
+
+class AddreessItemState extends State<AddreessItem>
+    with WidgetsBindingObserver {
+  bool isAddressCopied = false;
+  bool isAddressSelected = false;
+
+  @override
+  void initState() {
+    super.initState();
+    isAddressSelected = widget.account.address == widget.currentAddress;
+  }
+
+  _syncSpinnerOrBalanceDisplay(Account account, ThemeData theme) {
+    ExtendedTheme extendedTheme = theme.extension<ExtendedTheme>()!;
+    final isAddressSelected = account.address == widget.currentAddress;
+    final textStyle = isAddressSelected
+        ? extendedTheme.monoMediumText
+        : extendedTheme.monoRegularText;
+
+    return BlocBuilder<ExplorerBloc, ExplorerState>(
+        builder: (BuildContext context, ExplorerState state) {
+      if (state.status == ExplorerStatus.singleSync &&
+          state.data['address'] == account.address) {
+        return Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+            SizedBox(
+                height: 16,
+                width: 16,
+                child: CircularProgressIndicator(
+                  color: theme.textTheme.labelMedium?.color,
+                  strokeWidth: 2,
+                  value: null,
+                  semanticsLabel: 'Circular progress indicator',
+                ))
+          ],
+        );
+      } else {
+        return Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
+          Text('Received payments totalling',
+              textAlign: TextAlign.start,
+              style: textStyle!.copyWith(
+                  fontFamily: 'Almarai',
+                  fontWeight: FontWeight.w300,
+                  fontSize: 12)),
+          SizedBox(height: 4),
+          Text(
+              '${account.balance.availableNanoWit.standardizeWitUnits().formatWithCommaSeparator()} ${WIT_UNIT[WitUnit.Wit]}',
+              textAlign: TextAlign.start,
+              style: textStyle.copyWith(
+                  fontFamily: 'Almarai',
+                  fontWeight: FontWeight.w300,
+                  fontSize: 12)),
```

Comment:
Should we merge https://github.com/witnet/my-wit-wallet/pull/604 first to avoid setting this values here?